superset-frontend/src/components/Select/Select.tsx

```diff
@@ -315,81 +325,44 @@ const Select = ({
     : allowNewOptions
     ? 'tags'
     : 'multiple';
-  const allowFetch = !fetchOnlyOnSearch || searchedValue;
-
-  // TODO: Don't assume that isAsync is always labelInValue
-  const handleTopOptions = useCallback(
```

Comment:
`handleTopOptions` is not needed anymore since we are handling sorting selected options to the top also in the sort comparator.